atomic_reactor/plugin.py

```diff
@@ -354,6 +396,44 @@ def __init__(self, dt, workflow, plugins_conf, *args, **kwargs):
         self.plugins_results = workflow.prebuild_results
         super(PreBuildPluginsRunner, self).__init__(dt, workflow, 'PreBuildPlugin', plugins_conf, *args, **kwargs)
 
+
+class BuildStepPlugin(BuildPlugin):
+    pass
+
+
+class BuildStepPluginsRunner(BuildPluginsRunner):
+
+    def __init__(self, dt, workflow, plugin_conf, *args, **kwargs):
+        logger.info("initializing runner of build-step plugin")
+        self.plugins_results = workflow.buildstep_result
+
+        if not plugin_conf:
+            # if none buildstep plugins specified, fallback to docker api plugin
+            plugin_conf = [{'name': 'docker_api', 'is_allowed_to_fail': False}]
+        else:
```

Comment:
logic here is quite different

if no plugins specified we will fallback to docker_api, required isn't specified because it is default in pluginsrunner

else is there for case there are buildstepplugins specified BUT we want to control their required & is_allowed_to_fail regardless what they had set